catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OperationsMetacardSupport.java

```diff
@@ -147,22 +148,48 @@ void generateMetacardAndContentItems(
           fileName = updateFileExtension(mimeTypeRaw, fileName);
         }
 
-        Metacard metacard =
-            metacardFactory.generateMetacard(mimeTypeRaw, contentItem.getId(), fileName, tmpPath);
-        metacardMap.put(metacard.getId(), metacard);
+        TransformResponse transformResponse =
+            frameworkProperties
+                .getTransform()
+                .transform(
+                    new MimeType(mimeTypeRaw),
+                    contentItem.getId(),
+                    null,
+                    fileName,
+                    tmpPath.toFile(),
+                    null,
+                    arguments);
+
+        Optional<Metacard> parentMetacardOptional = transformResponse.getParentMetacard();
+        if (parentMetacardOptional.isPresent()) {
+          Metacard parentMetacard = parentMetacardOptional.get();
+          metacardMap.put(parentMetacard.getId(), parentMetacard);
+          contentItems.add(
+              new ContentItemImpl(
+                  parentMetacard.getId(),
+                  StringUtils.isNotEmpty(contentItem.getQualifier())
+                      ? contentItem.getQualifier()
+                      : "",
+                  com.google.common.io.Files.asByteSource(tmpPath.toFile()),
```

Comment:
Nit: Should this be statically imported?

Reply:
yes, because `java.nio.file.Files` is already imported, so the class names would otherwise conflict.